src/odemis/acq/acq_stream_settings.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 24 Jan 2024
+
+@author: Karishma Kumar
+
+Copyright © 2024-2025 Karishma Kumar, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import json
+import logging
+import os
+from datetime import datetime
+from typing import List, Dict, Any
+
+from odemis.gui.util import get_home_folder
+
+CONF_PATH = os.path.join(get_home_folder(), u".config/odemis")
+MAX_LENGTH = 10  # Maximum number most recent settings to display
+
+
+def get_config_index(config_data: list, name: str) -> int:
+    """
+    Get the index of the config entry with the given name
+    :param config_data: list of config entries
+    :param name: name of config entry
+    :return: index of config entry with given name
+    """
+    try:
+        index = list(map(lambda x: x["name"], config_data)).index(name)
+    except:
+        index = None
+
+    return index
+
+
+def add_and_sort_config_entries(config_data: list, new_entries: list, limit: int = 10) -> List[Dict]:
+    """
+    Add new entries to config_data and sort by last_updated, limit to 10
+    :param config_data: list of config entries
+    :param new_entries: list of new config entries
+    :param limit: limit of config entries
+    :return: sorted list of config entries
+    """
+
+    # add new entries to config_data
+    for entry in new_entries:
+        index = get_config_index(config_data, name=entry["name"])
+        entry["last_updated"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        if index is None:
+            config_data.append(entry)
+        else:
+            config_data[index] = entry
+
+    # sort by last_updated (timestamp)
+    sorted_config_data = sorted(config_data, key=lambda x: x["last_updated"], reverse=True)
+
+    return sorted_config_data[:limit]
+
+
+class AcquiredStreamSettings:
+    """
+    Read and write a list of most recently used (mru) settings from and to a JSON file respectively. The first element in the list
+    will be the latest used setting and the last element will be the oldest used setting. The list will consist of
+    finite number of entries which will update in the above mention order each time the list of settings is updated and
+    consequently pushing out the oldest entry from the list.
+    """
+    file_name = "acquired_stream_mru.json"
```

Comment:
Let's move this to the GUI. I'd suggest more like `stream_settings.json`, or `stream_settings_mru.json`